### DISCUSSION TITLE
Vaillant ecoTEC (UK) - Odd behaviour

### DISCUSSION THREAD

**Discussion by richardcastle** [john30/ebusd#869] [Q&A]
Kudos to John and all the contributors for such a wonderful initiative.

We have a dual set of Vaillant ecoTEC plus 428 boilers chained together using archaic AC binary signalling. (Thermostat and timerswich are AC driven feeds to the boiler chain). We wanted to start our journey towards better boiler control, so fitted some ebuzzz adapters and configured ebusd to work it's magic on a Pi providing data back to Home Assistant.

Data is coming through and the bus appears stable with minimal errors. Update rate per sensor seems to be in minutes, which based on how fast the activity LED is flashing seems very slow. Is that normal?

However my main query is whether simply wiring in two boards, (1 per boiler as recommended) will alter the operation of the boilers in any way? Our first intention was purely to monitor the data on the bus and once we built confidence, we would then start to look at building a controls schema.

The reason we ask is that the boilers appear to be short-cycling. Firing up. Running for a minute, then running through a overrun/anti-cycle routine for a few minutes before firing back up. Demand signal does not change. It's possible the issue started after I fitted the boards, so obvious first step was to remove the ebus connection and see if anything changed. Still a problem.

So my question to the wonderful folks here is whether adding an eBus reader to each boiler should cause the boilers to change state and are now getting confused by the presence of AC demand drive and the presence of something on the bus?

I've scoured the discussions and I'm thinking that it's more likely completely un-related, but wanted to make absolutely sure before the plumber blames my interfacing!

**Comment by G8GYW** [comment]
Hi Richard, I can confirm that attaching an adapter to the ebus will not affect the boiler's operation in any way unless you issue a command to change its parameters.

Short cycling is a feature of Vaillant 400 series boilers. I've spent many hours trying to prevent it on my 418 without success.

Mike

**Comment by richardcastle** [comment]
Thank you for the response Mike. That helps to get the confirmation.

The boilers have been in place for a decade now and the short-cycling issue has been a recent occurrence this winter. The eBus boards have been hugely helpful in giving us very visible data on the issue.

Right now it seems that there is an issue with the hot water circuit as flow temperature climbs very quickly on ignition, causing it to trip and the boiler to start on pump-overrun cycle. We get a grand total of 10 seconds of heating on that circuit before. When I add another circuit in, the whole system behaves. Most likely a pump under-performing or a stuck valve somewhere. Both boilers respond in the same way, so seems unlikely that both boilers have failed with the same fault!

I corrected the update rate issue by actively polling MQTT for updates on the specific messages. I didn't realise that was required.

Out of interest, is there any way to drive boiler demand via eBus control? I suspect it's possible as Vaillant have other eBus control units which can drive demand. I was hoping I could simply activate heating demand by posting an MQTT command and then vary power output using PartloadHcKW (which I'm already successfully able to do). It seems looking at the eBus command set for the 428, there isn't anything obvious there that controls demand. I'd dearly like to dump the AC signalling and build a controller system in Home Assistant.

**Reply by G8GYW** [reply]
My 418 was installed in 2009 but I have only recently started to monitor the ebus. There were a lot of complaints about short cycling in the early years but I don't know if Vaillant redesigned their software or control board to address it and I don't fancy forking out £300+ for a replacement to find out. I haven't tried driving demand but I have both the VRT392f and VRC470f controllers to play with and they allow you to set desired room and flow temperatures.

Mike

**Comment by Dinth** [comment]
@richardcastle could you let me know if you find out how to drive demand? 
I think i have previously seen someone trying to achieve it via Status01 and Status02 commands, but im not sure if it was resolved

**Reply by richardcastle** [reply]
I've found a few threads that suggest it is possible using SetMode or SetModeOverride which replicates the control messaging from an external eBus connected boiler controller. Unfortunately enabling the functionality requires modification of the eBus configuration files to manually add in the command. Not for the faint hearted!

Some useful links to read through here:

https://github.com/john30/ebusd-configuration/issues/170
https://github.com/john30/ebusd/issues/190
https://github.com/john30/ebusd/issues/179

**Reply by HiQual** [reply]
This link worked for me ...
https://github.com/john30/ebusd-configuration/issues/242#issuecomment-1194481837

**Comment by HiQual** [comment]
@richardcastle 
The following suggestion is not related to ebus(d).
It is related to short cycling.
Try going into the boiler engineers settings and changing "d0" from "auto" to say "10" on both boilers - just to see the effect on short cycling.
This is called "range rating" the boilers - you are limiting their maximum output so that they cannot race away and overshoot at the start of a heating cycle.
If this helps with the short cycling you can then increase the d0 setting to suit your requirements.

I have been looking for answers to find if there is any way to use ebusd to change the d0 (partloadhckw) setting - but without success.  It appears you can do so and I would be very grateful to hear how?
Hope that helps.

**Reply by G8GYW** [reply]
This won't work because Vaillant boilers fire up at 2/3 of the _maximum_ output, not the range rated output. My 418 fires up at 12kW even though d.0 is set to 8kW.

Changing d.0 is straightforward:

 ebusctl write -c bai partloadHcKW 8

Mike

**Reply by HiQual** [reply]
Sorry, but your comment is not universally true for all Vaillant boilers - range rating controls startup overshoot and startup short cycling on my current Vaillant Home 18 and on my previous Vaillant 412 ecoTec.  I can see from my ebusd generated charts that if I set d0 at 12Kw on my 24Kw boiler then the maximum modulation achieved is 50%
.
Unfortunately, changing d.0 is not straightforward for me.  I can read it, and I can send the write command with response "done" - but for me it is not actually changed on the boiler.

**Reply by G8GYW** [reply]
Fair enough, but my 2009 vintage ecoTEC plus 418 always fires up at 66% then modulates down to d.0 and finally settles at the boiler's minimum of 5kW.

Have you edited the boiler's .inc file to make partloadHcKW writable?

**Reply by HiQual** [reply]
Just checked, yes it's writable (r;wi) thanks.

**Comment by richardcastle** [comment]
Thanks for the comments!

So I've already been reducing the power output via D.0 (I was also able to change it from HA using MQTT as it sometimes resets back to 30). I have it set to 5kw and this has made no perceivable difference. However @G8GYW observations that the boilers fire at 2/3 max output seems to fit what I'm observing here (cycling still happens regardless of whether the boiler is set at 30 or 5). I get a huge peak on first ignition, the flow temperature hits 10% above dial temp, and the whole system shuts down and drops back into cycle mode.

@HiQual are there any other D.x variables that I could be missing here?

**Reply by HiQual** [reply]
Sorry I don't know of any other d.x variables that might be relevant.
As someone said earlier, it sounds like you have a lack of sufficient water flow through the boilers.

**Reply by richardcastle** [reply]
Yep. That was me. ;)
